adapters/cisco_nx_rest/cisco_nx_rest_command.php

```diff
@@ -160,23 +160,36 @@ function eval_DELETE() {
 		global $SMS_RETURN_BUF;
 
 		foreach ( $this->delete_list as $delete ) {
-			$operation = trim ( $delete->evaluate_operation () );
-			debug_dump ( $operation, "DELETE CONF\n" );
-			$xpath = trim ( $delete->evaluate_xpath () );
-			debug_dump ( $xpath, "DELETE XPATH\n" );
-
-			if (! empty ( $operation )) {
-                             $conf = $operation . '##' . $xpath;
-                             $xml_conf = trim($delete->evaluate_xml());
-                             $xml_conf_str = str_replace("\n", '', $xml_conf);
-				if (! empty ( $xml_conf_str )) {
-	                             $conf .= "' -d'".$xml_conf_str;
-				}
-
-                             $this->configuration .= "{$conf}\n";
-                             $SMS_RETURN_BUF .= "{$conf}\n";
+                        $operation_str = trim ( $delete->evaluate_operation () );
+                        $operations = explode ( "##", $operation_str );
+
+                        $xpath_str = trim ( $delete->evaluate_xpath () );
+                        $xpaths = explode ( "##", $xpath_str );
+
+                        $xml_conf_str = trim ( $delete->evaluate_xml () );
+                        $xml_conf_str = str_replace ( "\n", '', $xml_conf_str );
+                        $xml_configs = explode ( "##", $xml_conf_str );
+
+                        if (! empty ( $operation_str )) {
+                                 if (count ( $xpaths ) != count ( $operations )) {
+                                        throw new SmsException ( "Operations are not as many as Xpaths" );
+                                } else
+                                                                {
+                                  $i = 0;
+                                  foreach ( $operations as $operation )
+                                  {
+                                        $conf = $operation . '##' . $xpaths[0];
+                                        $xml_conf_str = str_replace("\n", '', $xml_conf[0]);
+                                        if (! empty ( $xml_conf_str )) {
+                                                 $conf .= "' -d'".$xml_conf_str;
+                                        }
+                                        $this->configuration .= "{$conf}\n";
+                                        $SMS_RETURN_BUF .= "{$conf}\n";
```

Comment:
$SMS_RETURN_BUF can be set outside of the loop :
$SMS_RETURN_BUF = $this->configuration;